subsys/bluetooth/host/conn.c

```diff
@@ -190,25 +190,74 @@ bool le_param_req(struct bt_conn *conn, struct bt_le_conn_param *param)
 	return true;
 }
 
+static int conn_le_param_update(struct bt_conn *conn,
+				const struct bt_le_conn_param *param)
+{
+	BT_DBG("conn %p features 0x%02x params (%d-%d %d %d)", conn,
+	       conn->le.features[0], param->interval_min,
+	       param->interval_max, param->latency, param->timeout);
+
+	/* Use LE connection parameter request if both local and remote support
+	 * it; or if local role is master then use LE connection update.
+	 */
+	if ((BT_FEAT_LE_CONN_PARAM_REQ_PROC(bt_dev.le.features) &&
+	     BT_FEAT_LE_CONN_PARAM_REQ_PROC(conn->le.features)) ||
+	    (conn->role == BT_HCI_ROLE_MASTER)) {
+		return bt_conn_le_conn_update(conn, param);
+	}
+
+	/* If remote master does not support LL Connection Parameters Request
+	 * Procedure
+	 */
+	return bt_l2cap_update_conn_param(conn, param);
+}
+
 static void le_conn_update(struct k_work *work)
```

Comment:
I find the 2 names `le_conn_update` and `conn_le_param_update` very confusing. Can you rename them to something that clearly shows what the functions do?